rpc/service.go

```diff
@@ -21,15 +21,22 @@ type RPCHandler interface {
 	AddOrders(orders []*zeroex.SignedOrder) (*zeroex.ValidationResults, error)
 	// AddPeer is called when the client sends an AddPeer request.
 	AddPeer(peerInfo peerstore.PeerInfo) error
-	// SubscribeToOrders is called when a client sends a Subscribe to orderStream request
+	// SubscribeToOrders is called when a client sends a Subscribe to `orders` request
 	SubscribeToOrders(ctx context.Context) (*rpc.Subscription, error)
+	// SubscribeToHeartbeat is called when a client sends a Subscribe to `heartbeat` request
+	SubscribeToHeartbeat(ctx context.Context) (*rpc.Subscription, error)
 }
 
-// Orders calls rpcHandler.Orders and returns the rpc subscription.
+// Orders calls rpcHandler.SubscribeToOrders and returns the rpc subscription.
 func (s *rpcService) Orders(ctx context.Context) (*rpc.Subscription, error) {
 	return s.rpcHandler.SubscribeToOrders(ctx)
 }
 
+// Heartbeat calls rpcHandler.SubscribeToHeartbeat and returns the rpc subscription.
+func (s *rpcService) Heartbeat(ctx context.Context) (*rpc.Subscription, error) {
```

Comment:
I think we should consider implementing `Heartbeat` directly in `rpcService` instead of making it part of the `RPCHandler` interface. The other methods require knowledge of the database, watchers, and other packages; it would break separation of concerns to implement them in `rpcService`. But the `Heartbeat` method doesn't depend on the database, watchers, or any other packages. `Heartbeat` is really just a concern of the `rpc` package itself. What do you think?

Reply:
Yep, makes sense. Moved it.